test/server/core/card/Upgrade.spec.js

```diff
@@ -105,5 +105,34 @@ describe('Upgrade cards', function() {
                 expect(this.tielnFighter).toBeInLocation('discard');
             });
         });
+
+        describe('When an upgrade is attached,', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['waylay'],
+                        groundArena: ['pyke-sentinel'],
+                    },
+                    player2: {
+                        hand: ['entrenched'],
+                        groundArena: ['wampa'],
+                    }
+                });
+            });
+
+            it('and a unit is returned to its owner\'s hand, the upgrade should be in the upgrade\'s owner\'s hand', function () {
```

Comment:
I think this is a typo, shouldn't it be "the upgrade should be in the upgrade's owner's **discard**"?

Reply:
You are correct sir

Reply:
Still says hand

Reply:
yeah sorry, had to merge the changes from main, then i stash applied and lost the change... it worked on my machine :P 
I'll have this fixed pronto